File: Solutions/Codeforces/miscellaneous/B_Squares_and_Cubes.java
// !: My original solution is @ "./B_Squares_and_Cubes.py".. I may or may not have used generative AI to translate it to Java haha.....
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        
        for (int i = 0; i < t; i++) {
            int n = scanner.nextInt();
            int sqrtN = (int) Math.floor(Math.sqrt(n));
            int cbrtN = (int) Math.floor(Math.cbrt(n));
            int cbrtSqrtN = (int) Math.floor(Math.cbrt(Math.sqrt(n)));
            int result = sqrtN + cbrtN - cbrtSqrtN;
            System.out.println(result);
        }

        scanner.close();
    }
}

File: Solutions/Codeforces/miscellaneous/B_Squares_and_Cubes.py
import math

for _ in range(int(input())):
	n = int(input())

	print(math.floor(math.sqrt(n)) + math.floor(math.cbrt(n)) - math.floor(math.cbrt(math.sqrt(n))))

# !: This problem fails to be read by the CF judge due to "import math" being unsupported ? (it was forced for my PIE solution). The canonical solution is to search the solution space. Check my Java solution (../B_Squares_and_Cubes.java).